lib/mayaUsd/undo/OpUndoItems.cpp

```diff
@@ -355,6 +357,53 @@ bool SelectionUndoItem::redo()
     return status == MS::kSuccess;
 }
 
+//------------------------------------------------------------------------------
+// UfeSelectionUndoItem
+//------------------------------------------------------------------------------
+
+UfeSelectionUndoItem::UfeSelectionUndoItem(const std::string& name, const Ufe::Selection& selection)
+    : OpUndoItem(name)
+    , _selection(selection)
+{
+}
+
+UfeSelectionUndoItem::~UfeSelectionUndoItem() { }
+
+void UfeSelectionUndoItem::select(
+    const std::string&    name,
+    const Ufe::Selection& selection,
+    OpUndoItemList&       undoInfo)
+{
+    auto item = std::make_unique<UfeSelectionUndoItem>(name, selection);
+    item->redo();
+    undoInfo.addItem(std::move(item));
+}
+
+void UfeSelectionUndoItem::select(const std::string& name, const Ufe::Selection& selection)
+{
+    select(name, selection, OpUndoItemList::instance());
+}
+
+bool UfeSelectionUndoItem::undo()
+{
+    invert();
+    return true;
+}
+
+bool UfeSelectionUndoItem::redo()
+{
+    invert();
+    return true;
+}
+
+void UfeSelectionUndoItem::invert()
+{
+    auto globalSn = Ufe::GlobalSelection::get();
+    auto previousSelection = *globalSn;
```

Comment:
Can the returned value ever be null?

Reply:
No, the UFE global selection is initialized in Maya's model class, and we assert that the result is non-null.